src/analysis/special_functions/pow.lean

```diff
@@ -678,166 +689,231 @@ lemma rpow_nat_inv_pow_nat {x : ℝ} (hx : 0 ≤ x) {n : ℕ} (hn : 0 < n) :
 have hn0 : (n : ℝ) ≠ 0, by simpa [pos_iff_ne_zero] using hn,
 by rw [← rpow_nat_cast, ← rpow_mul hx, inv_mul_cancel hn0, rpow_one]
 
-section prove_rpow_is_continuous
-
-lemma continuous_rpow_aux1 : continuous (λp : {p:ℝ×ℝ // 0 < p.1}, p.val.1 ^ p.val.2) :=
-suffices h : continuous (λ p : {p:ℝ×ℝ // 0 < p.1 }, exp (log p.val.1 * p.val.2)),
-  by { convert h, ext p, rw rpow_def_of_pos p.2 },
-continuous_exp.comp $
-  (show continuous ((λp:{p:ℝ//0 < p}, log (p.val)) ∘ (λp:{p:ℝ×ℝ//0<p.fst}, ⟨p.val.1, p.2⟩)), from
-    continuous_log'.comp $ continuous_subtype_mk _ $ continuous_fst.comp continuous_subtype_val).mul
-  (continuous_snd.comp $ continuous_subtype_val.comp continuous_id)
-
-lemma continuous_rpow_aux2 : continuous (λ p : {p:ℝ×ℝ // p.1 < 0}, p.val.1 ^ p.val.2) :=
-suffices h : continuous (λp:{p:ℝ×ℝ // p.1 < 0}, exp (log (-p.val.1) * p.val.2) * cos (p.val.2 * π)),
-  by { convert h, ext p, rw [rpow_def_of_neg p.2, log_neg_eq_log] },
-  (continuous_exp.comp $
-    (show continuous $ (λp:{p:ℝ//0<p},
-            log (p.val))∘(λp:{p:ℝ×ℝ//p.1<0}, ⟨-p.val.1, neg_pos_of_neg p.2⟩),
-     from continuous_log'.comp $ continuous_subtype_mk _ $ continuous_neg.comp $
-            continuous_fst.comp continuous_subtype_val).mul
-    (continuous_snd.comp $ continuous_subtype_val.comp continuous_id)).mul
-  (continuous_cos.comp $
-    (continuous_snd.comp $ continuous_subtype_val.comp continuous_id).mul continuous_const)
+lemma rpow_deriv_aux_of_pos {p : ℝ × ℝ} (hp : 0 < p.1) :
```

Comment:
You are really proving two different statements here. I understand that by putting one single lemma you are able to reuse four lines of the proof, but for ease of discoverability I think it would be really a better idea to split this to two lemmas (or maybe you should even unite the two `times_cont_diff` statements into one single lemma.

Reply:
Which two `times_cont_diff` statements are you talking about?

Reply:
The one at the end of `rpow_deriv_aux_of_pos` and the one at the end of `rpow_deriv_aux_of_neg`.